daemon/graphdriver/devmapper/deviceset.go

```diff
@@ -395,27 +425,18 @@ func (devices *DeviceSet) deviceFileWalkFunction(path string, finfo os.FileInfo)
 		hash = ""
 	}
 
-	dinfo := devices.loadMetadata(hash)
-	if dinfo == nil {
-		return fmt.Errorf("Error loading device metadata file %s", hash)
+	// Include deleted devices also as cleanup delete device logic
+	// will go through it and see if there are any deleted devices.
+	if _, err := devices.lookupDevice(hash); err != nil {
```

Comment:
you mentioned in the comment above the signature of `lookupDevice` that this function should always be called after locking the deviceSet. There is nothing in `deviceFileWalkFunction` or the call stack that locks the deviceSet, at least as far as I can tell. Should we call here to `lookupDeviceLock`? why do we need a separated function if that function should always be called with a lock held?